### DISCUSSION TITLE
How to get depth from GPU

### DISCUSSION THREAD

**Discussion by copark86** [raysan5/raylib#2035] [Q&A] [answered]
I know there is a function to download a rendered image from GPU

Image GetTextureData(Texture2D texture);   

But is there a way to download depth from GPU? I am going to use it for image processing (not related to OpenGL).

**Comment by raysan5** [comment, marked as answer]
@copark86 I'm afraid this is out-of-scope for raylib, it's a specific use-case. Still, it can be accomplished rendering the depth buffer using a shader and getting it. Here there is an issue explaining the process: https://github.com/raysan5/raylib/issues/1872

A brief summary:

 1. Use a `RenderTexture` with a depth **`Texture`** instead of `RenderBuffer` (you need to change `LoadRenderTexture()` function or use `rlgl` directly for that)
 2. Render your world
 3. Render `RenderTexture.depth` using a Shader to linearize the depth values of that texture.
 4. `LoadImageFromScreen()` should return the drawn depth texture linearized currently on the screen framebuffer

There is probably a simpler way to do that but not exposed by raylib, you should use OpenGL directly and call yourself `glGetTexImage()` requesting a `GL_DEPTH_COMPONENT` as `GL_FLOAT` but I think returned data won't be linear.

**Reply by copark86** [reply]
Thanks! That helped me a lot.

**Reply by TextBox75** [reply]
Question, how can I render RenderTexture.depth? I'm trying to render the depth buffer to do shadow mapping.

**Comment by chriscamacho** [comment]
I did a toon shader using the same technique using normals instead of depth, but the process is identical save for an simple tweak in the shader, take a look at bedroomcoders.co.uk
